### ISSUE TITLE
[publication_list] subgroup of publications within individual years

### ISSUE THREAD

**Issue by bishesh** [getnikola/plugins#262]
Can we create subgroups of publications within the each individual year subtitle ?
I would like the pages to be rendered sth like:

### Category A
pub1
pub2
...
pubN

### Category B
pub1
pub2
...
pubN

### Categor C
pub1
pub2
...
pubN

### Category A
...

**Comment by xuhdev** [comment]
I can look into this. Instead of merely adding this feature, I would like to come up with more generalized filtering commands. What do you think of putting "Category A/B/C" outside the years?

## Category A

### 2018
### 2017
...

## Category B

### 2018
### 2017

**Comment by bishesh** [comment]
Yes, that is also quite useful. If we can allow the user to choose the hierarchy (including year) that would be general, and awesome!

**Comment by xuhdev** [comment]
I plan to make the whole list a customizable template which users can specify. However, I'm really busy this month. I'v marked this work for next month. Are you good with using it as it is for now?

**Comment by bishesh** [comment]
Yes, I'm using the plugin as it is for now; it is already quite nice and easy to use. 
Having the customization next month will be awesome!
Thanks

**Comment by xuhdev** [comment]
@bishesh Sorry for the delay. Unfortunately, I got trapped into job hunting, which took way longer time than what I expected. I have to move this work to a place on calendar at which time I will have received a job offer. (I probably will have some free time at that  time.) For now, as a workaround, is it possible for you to put publications into different bibfiles, and format them as my previous comments for your institute? Please let me know if you need help. Again, sorry for the inconvenience.

**Comment by bishesh** [comment]
No worries, best wishes with the job hunting!
I've publications of different authors in different bibfiles, but I'm not sure which previous comments are you referring to for which format ?

**Comment by xuhdev** [comment]
I meant to have this form:

## Category A

### 2018
### 2017
...

## Category B

### 2018
### 2017


You can achieve this by simply group publications from different categories to different bib files, and use a few publication list directives:

```rst
Category A
----------
... publication_list: categorya.bib

Category B
----------
... publication_list: categoryb.bib
```

**Comment by bishesh** [comment]
All right. Well with many different authors having their bibs on different files, this is not so convenient, but still doable.
